### DISCUSSION TITLE
Unable to open .vsi files

### DISCUSSION THREAD

**Discussion by lanternx** [AICAN-Research/FAST-Pathology#104] [Q&A] [answered]
Even with test data, I got this report
"FAST exception caught in Qt event handler OpenSlide was unable to open the file: E:/Downloads/OS-1.vsi"

**Comment by andreped** [comment, marked as answer]
Hello, @lanternx :]

FastPathology does not support reading .vsi images. FastPathology can only open the images supported by [OpenSlide](https://openslide.org/) + ome.tiff.

I recommend converting images in VSI format to a more open and better supported format like Generic TIFF, which works with commonly libraries like OpenSlide, FAST, cuCIM and other libraries.

I made a WSI converter for just that, if you want to give that a go:
https://github.com/andreped/vsi2tif

Here is a simple tutorial demonstrating how to convert the VSI images to Generic TIFF:
https://github.com/andreped/vsi2tif/blob/main/notebooks/conversion_tutorial.ipynb

**Reply by lanternx** [reply]
Thank you🙂
